.github/workflows/release.yml

```diff
@@ -50,19 +50,21 @@ jobs:
       PIP_DISABLE_PIP_VERSION_CHECK: 1
 
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3
       with:
         fetch-depth: 50
         submodules: true
 
-    - uses: actions/setup-python@v2
+    - uses: actions/setup-python@v4
+      with:
+        python-version: 3.x
 
     - name: Build source distribution
       run: |
-        python -m pip install -U setuptools wheel pip
+        python -m pip install --upgrade setuptools wheel pip
```

Comment:
pip install has [command line options](https://pip.pypa.io/en/stable/cli/pip_install/#cmdoption-user) for:
    --upgrade
    --upgrade-strategy
    --user
    --use-pep517

In a workflow, it is better to be explicit about which of these four we are using.